### PR TITLE
Endpoint for code generation with structured output

### PR DESCRIPTION
**Pull request by alexandrudanpop**
Part of OPS-2103

## Testing Checklist

Check all that apply:

- [ ] I tested the feature thoroughly, including edge cases

- [ ] I verified all affected areas still work as expected

- [ ] Automated tests were added/updated if necessary

- [ ] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

If there are UI/UX changes, include at least one of the following:

- Screenshots
- Loom video
- Preview deployment link

### CODE REVIEW

File: packages/server/api/test/unit/ai/context-enrichment.service.test.ts [hunk 1]
```diff
@@ -7,10 +7,11 @@ import {
   PopulatedFlow,
```
Comment (alexandrudanpop) [inline packages/server/api/test/unit/ai/context-enrichment.service.test.ts:7]: it was in the wrong folder

File: packages/server/api/src/app/flags/dev-flags.service.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ async function getAll(): Promise<Flag[]> {
   flags = [
     {
       id: FlagId.CODE_WITH_AI,
-      value: false,
+      value: true,
```
Comment (alexandrudanpop) [inline packages/server/api/src/app/flags/dev-flags.service.ts]: to reset
